### DISCUSSION TITLE
Concurrent connections and timeout

### DISCUSSION THREAD

**Discussion by offstd** [appy-one/acebase#63] [Q&A]
Is there a timeout configuration parameter?
How many concurrent connections can the server sustain?
Did anyone test this?

Thanks in advance

**Comment by appy-one** [comment]
The max amount of connections your server is able to handle depends on:

- the maximum amount of open file handles your server has configured (1 per websocket connection)
- a hard limit of 65,536 per IP address (1 per websocket connection), but there will be other ports open on your server for other purposes so there will be less available for websockets
- the load your clients put on the server: with many concurrent connections, you will likely be needing more than 1 server to handle the load.

An AceBase server uses Socket.IO for the connections. There is no timeout, a client tries to connect until it is able to. A client's `ready` event will not be raised until it connects, unless a cache database is used: when disconnected, the `ready` event will then fire once the cache database is ready.
